### PR TITLE
ci: update npm publish version

### PR DESCRIPTION
**Pull request by ThibodeauJF**
https://coveord.atlassian.net/browse/KIT-1832

### CODE REVIEW

File: packages/samples/headless-react/package.json [hunk 1]
```diff
@@ -28,7 +28,6 @@
     "web-vitals": "1.1.2"
   },
   "scripts": {
-    "prepublishOnly": "exit 1",
```
Comment (olamothe) [inline packages/samples/headless-react/package.json:31]: Why remove this ? It was there so that no one/no script accidentally publishes it.

File: package.json [hunk 2]
```diff
@@ -15,7 +15,7 @@
     "cypress:test": "lerna run cypress:test",
     "commit": "git-cz",
     "bump:version": "node ./scripts/deploy/bump-version.js",
-    "npm:publish:alpha": "lerna run npm:publish:alpha",
+    "npm:publish:alpha": "lerna run --stream npm:publish:alpha",
```
Comment (olamothe) [inline package.json:18]: Why is this needed now ?

**Review by olamothe — APPROVED** [review]

File: package.json (hunk 2 above)
Comment (ThibodeauJF) [reply package.json:18]: To see the debug information

File: packages/samples/headless-react/package.json (hunk 1 above)
Comment (ThibodeauJF) [reply packages/samples/headless-react/package.json:31]: It's private now so prepublish won't be run
